### PR TITLE
Added Redux to project

### PR DESCRIPTION
**Pull request by Daf98**
- Used ducks pattern to structure folders
- Made books' actions, and reducer
- Made categories' actions, and reducer
- Created root reducer using configureStore

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Write action creators to add and remove books
+export const addBook = (book) => ({
+  type: 'BOOK_ADDED',
```
Comment (elisha2kyakpo1) [inline src/redux/books/books.js]: - [x] kindly add action types and then use them in your action creators, something like this: `const BOOK_ADDED = bookstore/books/ADD_BOOK`. Let's do the same to the `BOOK_REMOVED`

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      return [
+        ...state,
+        {
+          title: action.title,
+          author: action.author,
+          id: action.id,
+        },
+      ];
+    case 'BOOK_REMOVED':
+      return state.map((id) => id !== action.id);
```
Comment (elisha2kyakpo1) [inline src/redux/books/books.js]: - [x] For the remove book action creator, we should use ES6 filter() method which returns a new array without a book we want to remove

File: src/redux/home/home.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Write action creator to check state
+export const checkState = () => ({
+  type: 'STATUS_CHECKED',
```
Comment (elisha2kyakpo1) [inline src/redux/home/home.js]: - [x] Let's create an action type for this action creator as well. Something like this:
`const CHECK_STATUS = bookstore/categories/CHECK_STATUS`

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Daf98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The PR title and description looks professional 👍
- You have a descriptive readme file 👍
- No linter errors 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Daf98 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well wrritten PR description 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**